test/integration/fake_upstream.cc

```diff
@@ -872,24 +872,23 @@ void FakeUpstream::FakeListenSocketFactory::doFinalPreWorkerInit() {
 FakeRawConnection::~FakeRawConnection() {
   // If the filter was already deleted, it means the shared_connection_ was too, so don't try to
   // access it.
-  if (auto filter = read_filter_.lock(); filter != nullptr) {
+  if (read_filter_ != nullptr && read_filter_.use_count() > 1) {
```

Comment:
use_count is not safe for multi-threaded environments as far as I remember. weak_ptr is safe.